frontend/js/new.js

```diff
@@ -120,13 +133,100 @@ const changeNumCols = ({ target }) => {
 	makeTriangleGrid(positionGrid)
 }
 
+/**
+ * Onclick function for converting tetrahedrons to Nmesh file.
+ * I haven't stress tested it, but it should fail when we have around a couple 
+ * million tetrahedrons. More efficient approaches will have to be considered then.
+ * https://nmag.readthedocs.io/en/latest/finite_element_mesh_generation.html#nmesh-file-format
+ * @param {Event} e
+ */
+const downloadNmeshFile = (e) => {
+	$('#download-nmesh-file').innerHTML = 'Generating...'
+	// Delimiter to use for point -> key generation
+	const DELIM = ','
+
+	/**
+ 	* @param {Point} point
+ 	*/
+	const generateKeyFromPoint = (point) => `${point.x}${DELIM}${point.y}${DELIM}${point.z}`;
+
+	/** @type {Array<Point>} 
+	 * An array to store unique nodes (points).
+	 */
+	const nodes = []
+
+	/** 
+	 * @type {Map<string, number>} 
+	 * A map to track unique nodes by their stringified coordinates, mapping to a node index. 
+	 */
+	const nodeMap = new Map()
+
+	/** @type {number} 
+	 * The index to assign to the next unique node. 
+	 */
+	let nodeIndex = 0
+
+	/** @type {Array<Array<number>>} 
+	 * An array to store tetrahedrons as arrays of node indices.
+	 * The inner array is always of length 4 and contains the node index for each point.
+	 */
+	const simplices = []
+
+	// Generate nodes and unique indexes for the node map
+	tetrahedrons.forEach((tetrahedron) => {
+		tetrahedron.forEach((point) => {
+			const key = generateKeyFromPoint(point)
+			if(!nodeMap.has(key)) {
+				nodeMap.set(key, nodeIndex)
+				nodes.push(point)
+				nodeIndex++
+			}
+		})
+	})
+
+	// Generate simplices
+	tetrahedrons.forEach((tetrahedron) => {
+		const tetrahedronAsNodeIndexes = tetrahedron.map((point) => nodeMap.get(generateKeyFromPoint(point)))
+		simplices.push(tetrahedronAsNodeIndexes);
+	})
+
+	/** @type {string} The content of the `.nmesh` file as a string. */
+	let nmeshFileContent = '# PYFEM mesh file version 1.0\n'
+
+	// TODO: If we aren't including surfaces in our file, do we need to include it here?
+	nmeshFileContent += `# dim = 3 nodes = ${nodes.length} simplices = ${simplices.length} periodic = 0\n` 
```

Comment:
Not sure if we need it in the header either. Probably not since it's just a comment, but we'll have to see.